File: screenrecorder/src/main/java/com/sunplus/screenrecorder/media/glutils/IDrawer2D.java
package com.sunplus.screenrecorder.media.glutils;

/**
 * Created by w.feng on 2018/10/11
 * Email: fengweisb@gmail.com
 */
public interface IDrawer2D {
  void release();

  float[] getMvpMatrix();

  IDrawer2D setMvpMatrix(final float[] matrix, final int offset);

  void getMvpMatrix(final float[] matrix, final int offset);

  void draw(final int texId, final float[] texMatrix, final int offset);

  void draw(final ITexture texture);

  void draw(final TextureOffscreen offscreen);
}

File: screenrecorder/src/main/java/com/almosting/screenrecorder/media/glutils/IDrawer2D.kt
package com.almosting.screenrecorder.media.glutils

/**
 * Created by w.feng on 2018/10/11
 * Email: fengweisb@gmail.com
 */
interface IDrawer2D {
  fun release()
  fun getMvpMatrix(): FloatArray?
  fun setMvpMatrix(matrix: FloatArray?, offset: Int): IDrawer2D?
  fun getMvpMatrix(matrix: FloatArray?, offset: Int)
  fun draw(texId: Int, texMatrix: FloatArray?, offset: Int)
  fun draw(texture: ITexture?)
  fun draw(offscreen: TextureOffscreen?)
}